content/cloud/variables.md

```diff
@@ -0,0 +1,208 @@
+title = "Configuring Spin Application Variables and Secrets"
+date = "2023-06-05T16:40:01.672734Z"
+template = "cloud_main"
+enable_shortcodes = true
+[extra]
+url = "https://github.com/fermyon/developer/blob/main//content/cloud/variables.md"
+
+---
+
+- [Prerequisites](#prerequisites)
+- [Creating Our Spin Application](#creating-our-spin-application)
+- [Adding Variables to an Application Component](#adding-variables-to-an-application-component)
+- [Using Variables in a Spin Application](#using-variables-in-a-spin-application)
+- [Deploying the Application to Fermyon Cloud](#deploying-the-application-to-fermyon-cloud)
+
+Spin supports dynamic application variables. Instead of being static, their values can be updated without modifying the application, creating a simpler experience for rotating secrets, updating API endpoints, and more. 
+
+These variables are defined in a Spin application manifest (in the `[variables]` section) and then provided by a configuration provider. Locally with Spin, the configuration provider may be Vault for secrets or host environment variables. See the [dynamic configuration documentation](../spin/dynamic-configuration.md) to learn about configuring variables locally. These variables can also be set and updated for Spin applications in the Fermyon Cloud using the the `spin cloud variables` CLI.
+
+This tutorial will walk through creating a simple application that validates a password.
+
+## Prerequisites
+
+First, you need to have Spin installed on your computer. Please use the official Fermyon Cloud Quickstart to both [install](/cloud/quickstart#install-spin) Spin and also [log in](/cloud/quickstart#log-in-to-the-fermyon-cloud) to Fermyon Cloud.
+
+Next, since this example is written in Python, lets make sure we have the required tools installed for Python Spin applications. Spin CLI facilitates the creation of new Spin applications through the use of application [templates](/cloud/cli-reference#templates). The template we are interested in, for this tutorial, is the `http-py` template. Lets ensure its installed along with the `py2wasm` Spin plugin that enables compiling the Python app to Wasm:
```

Comment:
```suggestion
Next, since this example is written in Python, let's make sure we have the required tools installed for Python Spin applications. Spin CLI facilitates the creation of new Spin applications through the use of application [templates](/cloud/cli-reference#templates). The template we are interested in, for this tutorial, is the `http-py` template. Let's ensure its installed along with the `py2wasm` Spin plugin that enables compiling the Python app to Wasm:
```